lib/okta/plug/event_hook.ex

```diff
@@ -0,0 +1,117 @@
+if Code.ensure_loaded?(Plug) do
+  defmodule Okta.Plug.EventHook do
+    @moduledoc """
+    This Plug handles Okta Event Hooks.
+
+    It validates the incoming requests:
+      * https://developer.okta.com/docs/concepts/event-hooks/
+
+    And forward the command to the `Okta.Event.HookHandler`.
+
+    Also implements Event Hooks verifications
+      * https://developer.okta.com/docs/concepts/event-hooks/#verifying-an-event-hook
+
+    ### Usage
+
+    Using `Plug.Router`
+
+        forward "/okta/event-hooks", to: Okta.Plug.EventHook, [
+          event_handler: MyModule,
+          secret_key: "my secret key"
+        ]
+
+    Using in `Phoenix.Router`
+
+        forward "/slack", Okta.Plug.EventHook, [
+          event_handler: MyModule,
+          secret_key: "my secret key"
+        ]
+
+    ### Configuration
+
+    * `event_handler`: A module that implements `Okta.EventHookHandler` behaviour.
+    * `secret_key`: The secret key to validate the incoming requests.
+    """
+
+    @behaviour Plug
+
+    alias Plug.Conn
+
+    @allowed_methods ["GET", "POST"]
+
+    def init(opts), do: opts
+
+    def call(conn, opts) do
+      conn
+      |> valid_request_method?()
+      |> authorize?(conn, opts)
+      |> handle_request(conn, opts)
+    end
+
+    defp valid_request_method?(%{method: method}) do
+      method in @allowed_methods
+    end
+
+    defp authorize?(false, _, _), do: false
+
+    defp authorize?(_, conn, opts) do
+      secret_key = secret_key(opts)
+
+      conn
+      |> authorization_token()
+      |> verify(secret_key)
+    end
+
+    defp verify(token, token), do: true
+    defp verify(_, _), do: false
+
+    defp authorization_token(conn) do
```

Comment:
you may want to call this method something else than using the terminology `token`. makes it seem like it is an regular authorization token when in fact its just they key. maybe `request_key` ?